sample-apps/ruby-rails-sample-app/config/initializers/opentelemetry.rb

```diff
@@ -0,0 +1,23 @@
+require 'opentelemetry/sdk'
+require 'opentelemetry/exporter/otlp'
+require 'opentelemetry/instrumentation/all'
+
+OpenTelemetry::SDK.configure do |c|
+    c.service_name = 'ruby-sample-app'
+  
+    c.id_generator = OpenTelemetry::Propagator::XRay::IDGenerator
+    c.propagators = [OpenTelemetry::Propagator::XRay::TextMapPropagator.new]
+  
+    c.add_span_processor(
+      OpenTelemetry::SDK::Trace::Export::BatchSpanProcessor.new(
+        OpenTelemetry::Exporter::OTLP::Exporter.new(
+          endpoint: 'http://0.0.0.0:4318/v1/traces'
```

Comment:
could we rely on the env var `OTEL_EXPORTER_OTLP_ENDPOINT` instead of hard coding this? https://opentelemetry.io/docs/concepts/sdk-configuration/otlp-exporter-configuration/#otel_exporter_otlp_endpoint

You could setup this environment variable in a docker-compose file. For development you can set it inside the development.rb file. 
```
ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'http://0.0.0.0:4318/v1/traces'
```



If yes, I believe we don't need to add a new span_processor?

Reply:
This is a good one. I updated this